modules/generator/config.go

```diff
@@ -144,7 +145,10 @@ func (cfg *ProcessorConfig) Validate() error {
 	if err := cfg.LocalBlocks.Validate(); err != nil {
 		errs = append(errs, err)
 	}
-	if err := cfg.HostInfo.Validate(); err != nil {
```

Comment:
`RegisterFlagsAndApplyDefaults` and `.Validate()` is a pattern we follow so this makes it odd one out.

maybe we can keep the Validate method as is to keep this pattern and move the actual validation into the `Validate`